### PR TITLE
Fix OKE templates and add e2e test

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:

Fixe OKE templates to add custom cloud init and add e2e test

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: cloud/scope/managed_control_plane.go [hunk 1]
```diff
@@ -282,7 +282,12 @@ func (s *ManagedControlPlaneScope) isResourceCreatedByClusterAPI(resourceFreeFor
 
 // GetClusterName returns the name of the cluster
 func (s *ManagedControlPlaneScope) GetClusterName() string {
-	return s.OCIManagedControlPlane.Name
+	okeClusterName := s.OCIManagedControlPlane.Name
+	// oke cluster names are restricted to less than 32
+	if len(okeClusterName) > 31 {
```
Comment (joekr) [inline cloud/scope/managed_control_plane.go:287]: Should we log that we are doing this? or at least document it in the book?

Also should our admission webhook do this?

File: cloud/scope/managed_machine_pool.go [hunk 2]
```diff
@@ -199,6 +199,15 @@ func (m *ManagedMachinePoolScope) FindNodePool(ctx context.Context) (*oke.NodePo
 	return nil, nil
 }
 
+func (m *ManagedMachinePoolScope) getNodePoolName() string {
+	nodePoolName := m.OCIManagedMachinePool.GetName()
+	// oke nopdepool names are restricted to less than 32
+	if len(nodePoolName) > 31 {
+		return nodePoolName[0:31]
```
Comment (joekr) [inline cloud/scope/managed_machine_pool.go:206]: what about adding this to the webhook do this? or maybe have a single `handleLongOKEName` or some other single use util function? I feel like we might forget this check is being done in two places. Or at least I will probably forget 😆

File: test/e2e/managed_cluster_test.go [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+			CancelWatches:     cancelWatches,
+			IntervalsGetter:   e2eConfig.GetIntervals,
+			SkipCleanup:       skipCleanup,
+			AdditionalCleanup: additionalCleanup,
+			ArtifactFolder:    artifactFolder,
+		}
+		dumpSpecResourcesAndCleanup(ctx, cleanInput)
+	})
+
+	It("Managed Cluster - Simple", func() {
```
Comment (joekr) [inline test/e2e/managed_cluster_test.go:116]: should this be `[PRBlocking]`?

**Review by joekr — CHANGES_REQUESTED** [review]
Can you post the test results in the PR as well?

File: cloud/scope/managed_control_plane.go (hunk 1 above)
Comment (shyamradhakrishnan) [reply cloud/scope/managed_control_plane.go]: good call out, moved it to webhook

File: cloud/scope/managed_machine_pool.go (hunk 2 above)
Comment (shyamradhakrishnan) [reply cloud/scope/managed_machine_pool.go]: same as above

File: test/e2e/managed_cluster_test.go (hunk 3 above)
Comment (shyamradhakrishnan) [reply test/e2e/managed_cluster_test.go]: done

**Comment by shyamradhakrishnan** [comment]
unit test
```
?   	github.com/oracle/cluster-api-provider-oci	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta1	1.430s	coverage: 17.9% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/config	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/ociutil	0.478s	coverage: 25.0% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/scope	168.777s	coverage: 76.7% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base/mock_base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement/mock_computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine/mock_containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/controllers	28.251s	coverage: 69.0% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1	1.811s	coverage: 10.0% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/controllers	1.662s	coverage: 55.2% of statements
?   	github.com/oracle/cluster-api-provider-oci/feature	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/version	[no test files]
```

**Comment by shyamradhakrishnan** [comment]
e2e test
```
[SynchronizedAfterSuite] PASSED [0.000 seconds]
[SynchronizedAfterSuite]
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/e2e_suite_test.go:252
------------------------------
[SynchronizedAfterSuite] PASSED [132.406 seconds]
[SynchronizedAfterSuite]
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/e2e_suite_test.go:252

  Timeline >>
  STEP: Tearing down the management cluster @ 01/05/23 08:14:39.304
  << Timeline
------------------------------

Ran 6 of 22 Specs in 2682.865 seconds
SUCCESS! -- 6 Passed | 0 Failed | 0 Pending | 16 Skipped

You're using deprecated Ginkgo functionality:
```

**Review by joekr — APPROVED** [review]

**Pull request merged**